### DISCUSSION TITLE
jetson-orin-nano-devkit-nvme: why it only flash sqi in flash.xml.in? How can I update APP partition?

### DISCUSSION THREAD

**Discussion by fengyingchu0827** [OE4T/meta-tegra#1334] [Q&A] [answered]
Hi, 
I used jetson-orin-nano-devkit-nvme.conf to build the image and I found only spi section in the file flash.xml.in of demo-image-full-jetson-orin-nano-devkit-nvme-20230819032419.tegraflash.tar.gz.
Why it only flash sqi in flash.xml.in? How can I build a image that can flash APP partition?

Thank you very much.

[flash.xml.in.log](https://github.com/OE4T/meta-tegra/files/12391013/flash.xml.in.log)
[jetson-orin-nano-devkit-nvme.conf.log](https://github.com/OE4T/meta-tegra/files/12391017/jetson-orin-nano-devkit-nvme.conf.log)

**Comment by madisongh** [comment]
Use [initrd flashing](https://github.com/OE4T/meta-tegra/wiki/initrd-flashing-support) on the Orin NX or Orin Nano devices to flash to NVMe or USB storage.

**Comment by ichergui** [comment]
Hey @fengyingchu0827 
Which branch you are using ?
Make sure that you follow the instructions that Matt shared with you.

**Reply by fengyingchu0827** [reply]
Hi @ichergui 
I am using the branch of kirkstone. Thanks.

**Comment by fengyingchu0827** [comment]
Hi @madisongh @ichergui 
I used sudo ./initrd-flash to flash the device successfully, but I think only spi is flashed.
I read initrd-flash.xml in the file of demo-image-full-jetson-orin-nano-devkit-nvme.tegraflash.tar.gz, and I found there is only spi section.
Also, After sudo ./initrd-flash, I found my device still have old yocto OS environment.
The branch I used is kirkstone.

[log.initrd-flash.2023-08-22-09.56.44.txt](https://github.com/OE4T/meta-tegra/files/12403981/log.initrd-flash.2023-08-22-09.56.44.txt)
[initrd-flash.xml.txt](https://github.com/OE4T/meta-tegra/files/12403985/initrd-flash.xml.txt)

**Reply by ichergui** [reply, marked as answer]
Hey @fengyingchu0827 
You are right. Please add the following line to your `conf/local.conf`
```
TNSPEC_BOOTDEV = "nvme0n1p1"
```
we need to backport the following commit to `kirkstone`. Matt will do that shortly.
https://github.com/OE4T/meta-tegra/commit/db98637327a066fea2eb8dce6bbeee00d351133c

Try to rebuild the image with the changes above and let me know.
